src/utils/UrqlCombinedError.re

```diff
@@ -39,24 +37,26 @@ type extension;
 /* A simple binding to the GraphQL error type exposed by graphql-js. See:
     https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/graphql/error/GraphQLError.d.ts.
    */
+[@bs.deriving abstract]
 type graphqlError = {
-  .
-  "message": Js.Nullable.t(message),
-  "locations": Js.Nullable.t(locations),
-  "path": Js.Nullable.t(array(path)),
-  "nodes": Js.Nullable.t(array(astNode)),
-  "source": Js.Nullable.t(source),
-  "positions": Js.Nullable.t(positions),
-  "originalError": Js.Nullable.t(originalError),
-  "extensions": Js.Nullable.t(Js.Dict.t(extension)),
+  message: Js.Nullable.t(message),
+  locations: Js.Nullable.t(locations),
+  path: Js.Nullable.t(array(path)),
+  nodes: Js.Nullable.t(array(astNode)),
+  source: Js.Nullable.t(source),
+  positions: Js.Nullable.t(positions),
+  originalError: Js.Nullable.t(originalError),
+  extensions: Js.Nullable.t(Js.Dict.t(extension)),
 };
 
-type combinedError('a) = {
-  .
-  "networkError": Js.Nullable.t(Js.Exn.t),
-  "graphqlErrors": Js.Nullable.t(array(graphqlError)),
-  "response": Js.Nullable.t('a),
-};
+class type _combinedError =
+  [@bs]
+  {
+    pub networkError: Js.Nullable.t(Js.Exn.t);
+    pub graphqlErrors: Js.Nullable.t(array(graphqlError));
+    pub response: 'response;
```

Comment:
What type is response? Also, should it be `Js.Nullable.t('response)`?

Is this type the same as the `'response` in the graphql request? As it is right now it could lead to type safety issues as we can't constrain it to any type

Reply:
Yep, good call out. I'll investigate what that actually is on the `urql` side and see if we can type it properly.

Reply:
Ok, so the proper type for it ends up being `Js.Nullable.t(Fetch.response)` b/c what you get back from `urql` is the literal `fetch` response object or `undefined`. Happy we get to use `bs-fetch` for the API 🎉 Another thing I did was add `combinedErrorToRecord`. This basically converts `urql`'s `combinedError` to a record type for us, which gives us a slightly nicer API in my opinion (`e.networkError`, `e.graphqlErrors`, `e.response`). Let me your thoughts @gugahoa and @Schmavery!